shuup/xtheme/models.py

```diff
@@ -174,6 +179,98 @@ def __str__(self):
         return _("Theme configuration for %s") % self.theme_identifier
 
 
+@python_2_unicode_compatible
+class Font(models.Model):
+    shop = models.ForeignKey(
+        "shuup.Shop", verbose_name=_("Shop"), related_name="admin_fonts", on_delete=models.CASCADE)
+    name = models.CharField(max_length=128, verbose_name=_("name"), help_text=_("Font family name"))
+    css_value = models.CharField(max_length=128, verbose_name=_("Font Family CSS Property"), help_text=_("(Optional) Input css font family property directly. Ex: font-family: Roboto;"))
+
+    woff = FilerFileField(
+        verbose_name=_("WOFF Font"),
+        blank=True, null=True,
+        on_delete=models.SET_NULL,
+        related_name="typography_woff_fonts"
+    )
+    woff2 = FilerFileField(
+        verbose_name=_("WOFF2 Font"),
+        blank=True, null=True,
+        on_delete=models.SET_NULL,
+        related_name="typography_woff2_fonts"
+    )
+    ttf = FilerFileField(
+        verbose_name=_("TTF font"),
+        blank=True, null=True,
+        on_delete=models.SET_NULL,
+        related_name="typography_ttf_fonts"
+    )
+    svg = FilerFileField(
+        verbose_name=_("SVG font"),
+        blank=True, null=True,
+        on_delete=models.SET_NULL,
+        related_name="typography_svg_fonts"
+    )
+    eot = FilerFileField(
+        verbose_name=_("EOT font"),
+        blank=True, null=True,
+        on_delete=models.SET_NULL,
+        related_name="typography_eot_fonts"
+    )
+
+    class Meta:
+        verbose_name = _("Font Family")
+        verbose_name_plural = _("Font Families")
+        unique_together = ("shop", "name")
+
+    def __str__(self):
+        return self.name
+
+    def get_font_sources(self):
+        font_sources = []
+
+        if self.eot:
+            font_sources.append("url('%s?#iefix') format('embedded-opentype')" % self.eot.url)
+        if self.woff:
+            font_sources.append("url('%s') format('woff')" % self.woff.url)
+        if self.woff2:
+            font_sources.append("url('%s') format('woff2')" % self.woff2.url)
+        if self.ttf:
+            font_sources.append("url('%s') format('truetype')" % self.ttf.url)
+        if self.svg:
+            font_sources.append("url('%s#svgFontName') format('svg')" % self.svg.url)
+
+        return font_sources
+
+class AdminThemeSettings(models.Model):
+    shop = models.ForeignKey(on_delete=models.CASCADE, to="shuup.Shop", related_name="admin_themes_settings", null=True)
```

Comment:
make this `unique=True` and `null=False`

Reply:
✅